From 6339ed0c2da1dd977bee08ce3c228003ba341fd4 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 8 Feb 2016 15:07:29 +0200
Subject: [PATCH 0845/1240] doc: a3700: Add boot device notes into
 documentation

- Add explanation of boot device selection for A3700 targets
  to build.txt

Change-Id: Id24e6420b21f586835f20cff2c12a5dd1261f956
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27347
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 doc/mvebu/build.txt | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/doc/mvebu/build.txt b/doc/mvebu/build.txt
index 472b319..52da096 100644
--- a/doc/mvebu/build.txt
+++ b/doc/mvebu/build.txt
@@ -32,7 +32,7 @@ Build Instruction:
 	# make <defconfig_file>
 	For AP806 stand alone board use "make mvebu_apn806_defconfig"
 	For Armada 8040Rz board use "make mvebu_armada70x0_defconfig"
-	For ArmadaLP board use "make mvebu_armadalp_defconfig"
+	For Armada-3700 board use "make mvebu_armadalp_defconfig"
 7. Build the U-Boot (Armada-3700 builds require WTP tools installation -
    refer to the section "Tools installation for Armada-3700 builds"):
 	# make
@@ -51,6 +51,33 @@ Configuration update:
 	To update the U-Boot configuration and the SPL configuration, please refer to
 	doc/README.kconfig
 
+Boot device selection note:
+---------------------------
+	The Armada-3700 supports eMMC as the boot device on target.
+	The default boot device on this platform is SPI NOR flash.
+	In order to change the boot device to eMMC, set the following parameter
+	the default configuration:
+	CONFIG_MVEBU_MMC_BOOT=y
+	If "menuconfig" is used, navigate to the following location:
+	-> ARM architecture
+	  -> Flash for image
+	and select "eMMC flash boot".
+	The eMMC has 2 boot partitions (BOOT0 and BOOT1), which are numbered as
+	partition 1 and 2 in MMC driver. Partition number 0 is the user data partition.
+	The default build script uses BOOT0 (partition 1) for storing the boot image
+	and u-boot parameters.  For changing the default boot partition, edit the
+	TIM descriptor builder script scripts/builditm.sh.
+	The default value: "EMMCPART=0x00000001" can be changed to "EMMCPART=0x00000002"
+	in order to use eMMC BOOT1 partition as the boot device.
+	The boot partition can be written from Linux command line using the following
+	commands:
+	# sudo echo 0 > /sys/block/mmcblk0boot0/force_ro
+	# sudo dd if=u-boot-spl.img of=/dev/mmcblk0boot0
+	The first commands removes the write protection flag from the boot partition 0.
+	The same operation can be done for BOOT1 partition by replacing the partition
+	id in the above commands from mmcblk0boot0 to mmcblk0boot1.
+
+
 Tools installation for Armada-3700 builds:
 ------------------------------------------
 1. Extract the tools/wtp/WTPTP_TOOLS_x_x_xx.zip in a temporary directory
-- 
1.9.1

